Description: Fix sign extension problem in bsd_attr_list()
 In FreeBSD xattr names can be up to 255 bytes long.
 Reading the length byte in an attr list as a signed char
 will result in negative values.
 .
 Noticed by Peter Rutenbar in Sourcefor bug #659.
Author: Jose Quinteiro <github@quinteiro.org>
Bug: https://github.com/Netatalk/Netatalk/pull/127
Last-Update: 2020-09-25
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/libatalk/vfs/extattr.c
+++ b/libatalk/vfs/extattr.c
@@ -353,13 +353,13 @@
     }
 
     /* Convert from pascal strings to C strings */
-    len = list[0];
+    len = (unsigned char)list[0];
     memmove(list, list + 1, list_size - 1);
 
     for(i = len; i < list_size; ) {
         LOG(log_maxdebug, logtype_afpd, "len: %d, i: %d", len, i);
 
-        len = list[i];
+        len = (unsigned char)list[i];
         list[i] = '\0';
         i += len + 1;
     }
